src/rules/no_unreachable.rs

```diff
@@ -35,6 +35,81 @@ impl LintRule for NoUnreachable {
       ProgramRef::Script(ref s) => visitor.visit_script(s, &DUMMY_NODE),
     }
   }
+
+  fn docs(&self) -> &'static str {
+    r#"Disallows the unreachable code after the control flow statements.
+
+Because the control flow statements(`return`,`throw`, `break` and `continue`) unconditionally exit a block of code, any statements after them cannot be executed.
+
+### Invalid:
+```typescript
+function foo() {
+  return true;
+  console.log("done");
+}
+```
+
+```typescript
+function bar() {
+  throw new Error("Oops!");
+  console.log("done");
+}
+```
+
+```typescript
+while (value) {
+  break;
+  console.log("done");
+}
+```
+
+```typescript
+throw new Error("Oops!");
+console.log("done");
+```
+
+```typescript
+function baz() {
+  if (Math.random() < 0.5) {
+    return;
+  } else {
+    throw new Error();
+  }
+  console.log("done");
+}
+```
+
+```typescript
+for (;;) {}
+console.log("done");
+```
+
+### Valid
+
+```typescript
+function foo() {
+  return bar();
+  function bar() {
+    return 1;
+  }
+}
+```
+
+```typescript
+function bar() {
+  return x;
+  var x;
+}
+```
+
+```typescript
+switch (foo) {
+  case 1:
+    break;
+    var x;
+}
+```"#
```

Comment:
The reason why `var x;` in these examples are valid may be quite unclear to people who are not so experienced. Maybe it's great to add some comment, like...

```suggestion
```typescript
function bar() {
  return x;
  // Because it's declared with `var`, this `x` gets hoisted.
  var x;
}
```

```typescript
switch (foo) {
  case 1:
    break;
    // Because it's declared with `var`, this `x` gets hoisted.
    var x;
}
```"#
```

(oh the suggestion is rendered awfully due to nested code blocks...)

But in this case, I would rather remove these two examples, because it's so rare that we use `var` declarations in today's JavaScript/TypeScript. As everyone knows, we should always use `let` or `const` instead.

Reply:
True. These two examples are not required.(also, support for ie11 is going to be over....)
I'll remove them.